File: Final/computer-vision/std/std.py
import queue


def bfs(maps, start_x, start_y):
    q = queue.Queue()
    q.put((start_x, start_y))
    maps[start_x][start_y] = 1
    dx = [0, 0, 1, -1]
    dy = [1, -1, 0, 0]
    while not q.empty():
        x, y = q.get()
        for i in range(4):
            nx = x + dx[i]
            ny = y + dy[i]
            if 0 <= nx < len(maps) and 0 <= ny < len(maps[nx]) and maps[nx][ny] == 1:
                maps[nx][ny] = 0
                q.put((nx, ny))


if __name__ == "__main__":
    n, m = map(int, input().split())
    maps = [[1] * (m + 2)]
    for i in range(n):
        maps.append([1] + list(map(int, input())) + [1])
    maps.append([1] * (m + 2))
    ans = 0
    for i in range(1, n + 1):
        for j in range(1, m + 1):
            if maps[i][j] == 1:
                bfs(maps, i, j)
                ans += 1
    print("CAB"[ans - 1])

File: Final/std/A/std.cpp
#include <queue>
#include <cassert>
#include <iostream>

constexpr int maxn = 505;

const int dx[] = {0, 0, 1, -1};
const int dy[] = {1, -1, 0 ,0};

int n, m;
int mp[maxn][maxn];

int bfs(int x, int y) {
    std::queue<std::pair<int, int>> q;
    q.emplace(x, y);
    mp[x][y] = 1;
    int siz = 0;
    while (!q.empty()) {
        auto [x, y] = q.front();
        siz++;
        q.pop();
        for (int i = 0; i < 4; ++i) {
            int nx = x + dx[i], ny = y + dy[i];
            // A little trick here: we add 0s around the map, so that all outer 0s are connected to each other.
            if (nx < 0 || nx > n + 1 || ny < 0 || ny > m + 1 || mp[nx][ny] || mp[nx][ny]) {
                continue;
            }
            mp[nx][ny] = 1;
            q.emplace(nx, ny);
        }
    }
    return siz > 20;
}

int main() {
    std::ios::sync_with_stdio(false);
    std::cin.tie(nullptr);
    std::cout.tie(nullptr);
    std::cin >> n >> m;
    for (int i = 1; i <= n; ++i) {
        std::string s;
        std::cin >> s;
        for (int j = 1; j <= m; ++j) {
            if (s[j - 1] == '0') {
                mp[i][j] = 1;
            }
        }
    }
    int ans = 0;
    for (int i = 0; i <= n + 1; ++i) {
        for (int j = 0; j <= m + 1; ++j) {
            if (mp[i][j]) {
                continue;
            }
            ans += bfs(i, j);
        }
    }
    std::cout << "CAB"[ans - 1] << '\n';
    return 0;
}
